test/utils_tests.js

```diff
@@ -37,4 +37,143 @@ describe('Utils', function () {
                           'The graph has a cycle.');
         });
     });
+
+    describe('searchAttributes', function () {
+        it('search with no args', function () {
+            const doc = staticDom(`
+                <img id= "foo" alt= "boo"></img><img id="fat" src= "bat"></img>
+            `);
+            const rules = ruleset(
+                rule(dom('img'), type('attr')),
+                rule(type('attr'), score(scoreFunc)),
+                rule(type('attr').max(), out('best'))
+            );
+
+            function scoreFunc(fnode) {
+                return searchAttributes(fnode, searchFunc)? 5 : 1;
+            }
+
+            function searchFunc(attr) {
+                return attr.includes('oo');
+            }
+            const facts = rules.against(doc);
+            const best = facts.get('best');
+            assert.equal(best.length, 1);
+            assert.equal(best[0].element.id, 'foo');
+        });
+
+        it('search with args', function () {
+            const doc = staticDom(`
+                <img id= "foo" alt= "bat"></img><img id="sat" src= "bat"></img>
+            `);
+            const rules = ruleset(
+                rule(dom('img'), type('attr')),
+                rule(type('attr'), score(scoreFunc)),
+                rule(type('attr').max(), out('best'))
+            );
+
+            function scoreFunc(fnode) {
+                return searchAttributes(fnode, searchFunc, 'id')? 5 : 1;
+            }
+
+            function searchFunc(attr) {
+                return attr.includes('at');
+            }
+            const facts = rules.against(doc);
+            const best = facts.get('best');
+            assert.equal(best.length, 1);
+            assert.equal(best[0].element.id, 'sat');
+        });
+
+        it('search array attribute', function () {
+            const doc = staticDom(`
+                <img id = "fat" class="fat bat sat" ></img><img id ="foo" class="foo bar boo"></img>
```

Comment:
As a small nit, the spacing around the equals signs in these tests is inconsistent. It would be good to be consistent about that.